src/replication.c

```diff
@@ -888,6 +899,7 @@ update_replica_entry(spec_t *spec, replica_t *replica, int iofd)
 		REPLICA_ERRLOG("pthread_create(r_thread) failed for "
 		    "replica(%lu)\n", replica->zvol_guid);
 replica_error:
+		destroy_mempool(&replica->cmdq);
```

Comment:
at various places, there is 'goto' to label 'replica_error' even before cmdq is initialized.